### DISCUSSION TITLE
Having Trouble Implementing Layered Fields for Novel Writing

### DISCUSSION THREAD

**Discussion by savvyslog** [SkepticMystic/breadcrumbs#561] [Q&A]
I love the idea of this plugin and want to use it to keep track of the different parts of my novels.

## What I Want to Do
I'm trying to implement the following hierarchy to show a section (in the matrix view) for each other type in each note:

- Book
	- Part
		- Chapter
			- Scene

I.e., a Book would show the Parts, Chapters, and Scenes it contains, a Part would show the Book it is part of as well as the Chapters and Scenes it contains, etc.

Ideally this would all be calculated via the frontmatter I already use for each note type. That would allow me to have clickable links to the various parts, especially in Dataview tables and on mobile.

## Things I've Tried
I've been able to get parts of it to work, but not all the layers, and certainly not all together. I've added each as an edge field and put them in corresponding groups. This gets some of entries working (via my frontmatter links), but not all of them.

I assume (though could be wrong) there should be a way to get this to work using the Transitive Implied Relations settings, but I haven't been able to get that to work completely. Namely, I haven't been able to find a combination of rules that include only the desired results in each section. 

For example, if I try to connect the Chapter and Book fields/notes, it then also includes all related (ie, notes with links to that Book note in their frontmatter) Scenes and Parts notes as well.

I also tried using the Dataview Notes option, with the query in the frontmatter. However, I can't figure out if there's a way to add a "WHERE" clause, so I could limit the chapters to just those in a specific book, etc. I know I could put each note type in their own (sub)folder and query them that way, but I'd prefer a flat folder hierarchy.

I also tried the Johnny Decimal Notes option, which seemed to work okay, but I don't see a way to specify the specific fields to be added — only a single choice for all. Plus it would require me to decide on the exact literal hierarchy upon note creation, which can be restrictive. 

I also tried the List Note option, which works well, but requires me to manually update the list upon any change. This compounds when I want to have Chapters link up to both Parts and Books, which I can do with this option, but requires me to list each Chapter twice, once under Book and once under Part. 

## Questions
So in the end, I guess my questions are:
- Is there a way to use the Transitive Implied Relations settings to setup the individual relationships like I am looking for?
- Is there a way to use Dataview "WHERE" clauses within the query placed in the frontmatter of Dataview Notes?
- Am I just trying to do something that the plugin wasn't intended to do?

**Comment by michaelpporter** [comment]
Great use case — this is well within what Breadcrumbs does; the friction is that **transitive rules are global**. A rule like `up → down` applies to *every* field, so if all four layers share `up`/`down`, one rule chains Book↔Part↔Chapter↔Scene together and dumps them in a single section. The fix is to give **each layer-pair its own dedicated field** — the Matrix groups by field, so distinct fields = distinct sections with no bleed.

Put one upward typed link per ancestor on each note:

- `Scene.md`: `scene-of-chapter:: [[Chapter]]`
- `Chapter.md`: `chapter-of-part:: [[Part]]`, `chapter-of-book:: [[Book]]`
- `Part.md`: `part-of-book:: [[Book]]`

Then add a **reverse (close) transitive rule per field** so parents show children in their own sections:

- chain `[scene-of-chapter]` → close `chapter-has-scene`
- chain `[chapter-of-part]` → close `part-has-chapter`
- chain `[chapter-of-book]` → close `book-has-chapter`
- chain `[part-of-book]` → close `book-has-part`

Now a Book shows `book-has-part` / `book-has-chapter` / (`book-has-scene`) as separate sections, a Chapter shows its Book/Part above and its Scenes below — all clickable frontmatter links. The trade-off: you list each ancestor explicitly (a Chapter names both its Part and Book). You can auto-derive skip-level links (e.g. Book→Scene) with a multi-field chain like `[book-has-chapter, chapter-has-scene]` → close `book-has-scene`, keeping it in its own section.

On your other tries:

- **Dataview Notes + `WHERE`**: not supported — `BC-dataview-note-query` takes a Dataview *source* (tags/folders/`outgoing([[X]])`/`and`/`or`/`-`), not a full query, so there's no `WHERE` clause. The dedicated-field approach above replaces it.
- **Johnny.Decimal / List notes**: both work but, as you found, force one field for all or manual upkeep — the typed-field + per-field reverse rules give you the precise per-layer sections without that.
